### PR TITLE
Updating the path param name to "namespace" instead of "namespaces"

### PR DESCRIPTION
**Pull request by nikhiljindal**
As per https://github.com/GoogleCloudPlatform/kubernetes/pull/9970#issuecomment-113025084

### CODE REVIEW

**Comment by nikhiljindal** [comment]
cc @bgrant0607

**Comment by bgrant0607** [comment]
Thanks, LGTM. Please make shippable green.

**Comment by bgrant0607** [comment]
Lots of namespace errors -- looks like legit failures.

**Comment by nikhiljindal** [comment]
Fixed tests!

**Comment by nikhiljindal** [comment]
Rebased.
Adding @lavalamp as second reviewer, since I need 2 LGTMs now :)

File: pkg/api/meta/interfaces.go [hunk 1]
```diff
@@ -115,10 +115,8 @@ type RESTScope interface {
 	// ParamName is the optional name of the parameter that should be inserted in the resource url
 	// If empty, no param will be inserted
 	ParamName() string
-	// ParamPath is a boolean that controls how the parameter is manifested in resource paths
-	// If true, this parameter is encoded in path (i.e. /{paramName}/{paramValue})
-	// If false, this parameter is encoded in query (i.e. ?{paramName}={paramValue})
-	ParamPath() bool
+	// ParamValueName is the optional name that should be used for the variable holding the value.
+	ParamValueName() string
```
Comment (lavalamp) [inline pkg/api/meta/interfaces.go]: Hm, what about "ArgumentName() string"? Clearer or less clear?

File: pkg/apiserver/api_installer.go [hunk 2]
```diff
@@ -250,7 +250,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	actions := []action{}
 
 	// Get the list of actions for the given scope.
-	if scope.Name() != meta.RESTScopeNameNamespace {
+	if scope.Name() == meta.RESTScopeNameRoot {
```
Comment (lavalamp) [inline pkg/apiserver/api_installer.go]: nit: if you make any other changes, can you convert this to a switch statement?

File: pkg/api/meta/interfaces.go (hunk 1 above)
Comment (nikhiljindal) [inline pkg/api/meta/interfaces.go]: Both seem fine to me.
I can change it if you think ArgumentName is better.

Comment (nikhiljindal) [inline pkg/api/meta/interfaces.go]: Renamed

File: pkg/apiserver/api_installer.go (hunk 2 above)
Comment (nikhiljindal) [inline pkg/apiserver/api_installer.go]: Done

**Comment by nikhiljindal** [comment]
Updated code as per comments.
PTAL.

**Comment by lavalamp** [comment]
LGTM

**Comment by bgrant0607** [comment]
LGTM, thanks.

**Pull request merged**